components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -0,0 +1,186 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.service.nimbus
+
+import android.content.Context
+import android.content.pm.PackageInfo
+import android.content.pm.PackageManager
+import android.os.Build
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import mozilla.components.service.glean.Glean
+import mozilla.components.support.base.coroutines.Dispatchers
+import mozilla.components.support.base.log.Log
+import mozilla.components.support.locale.getLocaleTag
+import org.mozilla.experiments.nimbus.AppContext
+import org.mozilla.experiments.nimbus.AvailableRandomizationUnits
+import org.mozilla.experiments.nimbus.EnrolledExperiment
+import org.mozilla.experiments.nimbus.RemoteSettingsConfig
+import org.mozilla.experiments.nimbus.NimbusClient
+import java.io.File
+import java.util.Locale
+
+/**
+ * This is the main experiments API, which is exposed through the global [Nimbus.shared] object.
+ */
+open class Nimbus internal constructor() {
+    companion object {
+        internal const val LOG_TAG = "service/Nimbus"
+        private const val EXPERIMENT_COLLECTION_NAME = "nimbus-mobile-experiments"
+        internal const val NIMBUS_DATA_DIR: String = "nimbus_data"
+
+        val shared by lazy { Nimbus() }
+    }
+
+    private val scope: CoroutineScope = CoroutineScope(Dispatchers.Cached)
+
+    private lateinit var nimbus: NimbusClient
+    private lateinit var dataDir: File
+    private var onExperimentUpdated: ((List<EnrolledExperiment>) -> Unit)? = null
```

Comment:
Spoke offline with Travis, it might make sense to use the [`ObserverRegistry`](https://github.com/mozilla-mobile/android-components/blob/5cd371eebc9e2dfef8cc45472dd52a72b5a36558/components/support/base/src/main/java/mozilla/components/support/base/observer/ObserverRegistry.kt) here to have consumers observe changes where needed.

An example implementation of this can be seen on the [`TopSitesStorage`](https://searchfox.org/mozilla-mobile/rev/07e55aadfee45db922078910fdb1c0691899f9c6/android-components/components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/DefaultTopSitesStorage.kt#32).

Reply:
@jhugman is writing up what a Nimbus observerable would look like for an iteration on this component in the milestone (M2) we are just kicking off.  I'd like to leave this as is for now to be able to close out M1. For our purposes in the first few experiments, a single observer is sufficient.